### PR TITLE
Upgrade to kotlin 1.8.0

### PR DESCRIPTION
**Pull request by gayanper**
Upgrade to kotlin 1.8.0

Seems we still cannot remove usage of deprecated ComponentRegistar at the moment, So added suppression for now which we can handle in 1.9.0 if the `org.jetbrains.kotlin.compiler.plugin.K2PluginRegistrar` is introduced as mentioned in https://youtrack.jetbrains.com/issue/KT-52665

### CODE REVIEW

**Comment by themkat** [comment]
Tried your change, and it seems to fix some issues like those we experience in #413 and https://github.com/fwcd/vscode-kotlin/issues/118. That being said, it causes a few new issues I can't pinpoint. Maybe it is something weird on my machine though. 

The error is `Cannot access 'java.io.Serializable' which is a supertype of 'kotlin.Int'. Check your module configuration`


To reproduce:
```kotlin
1 + 1
// ...or...
for (i in 1..10) {
}
```

**Comment by gayanper** [comment]
> Tried your change, and it seems to fix some issues like those we experience in #413 and [fwcd/vscode-kotlin#118](https://github.com/fwcd/vscode-kotlin/issues/118). That being said, it causes a few new issues I can't pinpoint. Maybe it is something weird on my machine though. Tried different types of project (barebones, Gradle etc.) both older and new (Kotlin 1.8.0).
> 
> The error is `Cannot access 'java.io.Serializable' which is a supertype of 'kotlin.Int'. Check your module configuration`
> 
> To reproduce:
> 
> ```kotlin
> 1 + 1
> // ...or...
> for (i in 1..10) {
> }
> ```

I think this could be cause by that the compiler environment doesn't have java runtime classes available. I will investigate and see.

But please provide feedback if there is anyone who has more knowledge in this area where things can be speed up.

**Comment by gayanper** [comment]
@themkat Please try to latest changeset. I have configured the compiler configuration to include the jvm classpaths as well after looking at the configuration API

**Review by themkat — APPROVED** [review]
@gayanper , it seems to work as expected now 😄 Awesome work upgrading! Hopefully it fixes all, if not most, of the issues I have linked 🙂

**Comment by gayanper** [comment]
@themkat fixed the tests as well.

**Comment by ilvez** [comment]
Thanks. Tested it out from your branch, got my project working at least :D

**Comment by theoparis** [comment]
Works for me too 👍🏼 However it seems like either a new PR should be created or this should be updated to support Kotlin [1.8.10](https://github.com/JetBrains/kotlin/releases/tag/v1.8.10)

**Comment by gayanper** [comment]
I can try updating the kotlin version tomorrow.

**Comment by gayanper** [comment]
Updated

**Comment by gayanper** [comment]
@themkat when can we merge this ? Seems some workflows are pending for approval by a maintainer.

**Comment by themkat** [comment]
@gayanper only @fwcd can merge and do anything maintainer related. I'm just someone who helps out when I have time :/

**Comment by gayanper** [comment]
> @gayanper only @fwcd can merge and do anything maintainer related. I'm just someone who helps out when I have time :/ Everything builds when I run them in both Java 11 and 17 locally, so should not be a problem. @fwcd , do you have time to look at this PR soon? It would help a lot of people with issues with newer versions of Kotlin 🙂

May be we should have more committers show using the language server daily and provide bugfixes to help @fwcd and secure the development of this project.

**Comment by fwcd** [comment]
Sorry for the delay, looks good to me, I'll merge this once CI passes.

**Comment by fwcd** [comment]
Thanks!

**Pull request merged**
